src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -445,6 +446,15 @@ private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext, bool hasUnsuppo
                                 continue;
                             }
 
+                            if (rule == s_removeUnreadMembersRule)
+                            {
+                                var dataMemberAttribute = symbolEndContext.Compilation.GetTypeByMetadataName(typeof(DataMemberAttribute).FullName);
+                                if (member.GetAttributes().Any(attr => attr.AttributeClass.Equals(dataMemberAttribute)))
+                                {
+                                    continue;
+                                }
+                            }
```

Comment:
i don't like this approach.  it will mask real problems with code.  if someone wants to state that their code is just using DataMember only to export data for external consumption, i think they should do so with an explicit suppression in their code.  Then the code is self documetning that they are purposefully doing this.